tests/functional/shutdown/23-scan-killed.t

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env bash
+# THIS FILE IS PART OF THE CYLC WORKFLOW ENGINE.
+# Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+# https://github.com/cylc/cylc-flow/pull/5196
```

Comment:
I did my best to avoid resorting to a functional test...

Reply:
I think this is suitable for integration testing with no sleeping or killing necessary.

```python
def test_scan_stopped(flow, scheduler, start, one_conf, test_dir):
    id_ = flow(one_conf)
    schd = scheduler(id_)
    async with start(schd):
        # the workflow is now running
        <copy contact => contact.bk>
    # the workflow is now stopped
    <copy contact.bk contact>
    assert await scan(states={'running', 'paused', 'stopping'}, scan_dir=test_dir, ping=True) == []
    assert len(await scan(states={'stopped'}, scan_dir=test_dir, ping=True)) == 1
```

Reply:
There's an existing test which does pretty much the same thing only it is testing `ClientError` (by making the server unresponsive) rather than `WorkflowStopped` (by copying the contact file):

https://github.com/cylc/cylc-flow/blob/79721142e872fb6fde97a79ef7a6f05e9332384c/tests/integration/test_scan.py#L384-L416

Reply:
I'll see if that works for me.

Reply:
I'm afraid I just can't get this working.